pkg/gcp/client.go

```diff
@@ -153,6 +160,10 @@ func (c *gcpClient) CreateWorkloadIdentityPool(ctx context.Context, parent, pool
 	return c.iamService.Projects.Locations.WorkloadIdentityPools.Create(parent, pool).WorkloadIdentityPoolId(poolID).Context(ctx).Do()
 }
 
+func (c *gcpClient) CreateWorkloadIdentityProvider(ctx context.Context, parent, providerID string, provider *iam.WorkloadIdentityPoolProvider) (*iam.Operation, error) {
+	return c.iamService.Projects.Locations.WorkloadIdentityPools.Providers.Create(parent, provider).WorkloadIdentityPoolProviderId(providerID).Context(ctx).Do()
```

Comment:
Can we add the contextWithTimeout() setup/teardown like the other wrapper functions?